src/HttpLib/protocol/UDP/PseudoTCPMessage.java

```diff
@@ -53,7 +60,39 @@ private void createPackets() {
         }
     }
 
+    public void addPacket(PseudoTCPPacket packet, int index) {
+        growPackets(index);
+        _packets.set(index, packet);
+    }
+
+    private void growPackets(int index) {
+        _packets.ensureCapacity(index);
+        while (_packets.size() <= index)
+            _packets.add(new PseudoTCPPacket());
+
+    }
+
     public PseudoTCPPacket[] getPackets() {
-        return _packets.toArray(new PseudoTCPPacket[_packets.size()]);
+        return _packets.toArray(new PseudoTCPPacket[0]);
+    }
+
+    public String getPeerAddress() {
+        return _peerAddress;
+    }
+
+    public int getPeerPort() {
+        return _peerPort;
+    }
+
+    public byte[] getPeerAddressBytes() {
+        return _peerAddressBytes;
+    }
+
+    public byte[] getPeerPortBytes() {
+        return _peerPortBytes;
+    }
+
+    public byte[] getPayload() {
+        return _payload;
```

Comment:
I thought that after adding the received packets back to the message object, the payload could be extracted as reassembly of the packet's payload. How's job would it be then?

Reply:
Call buildPayloadFromPackets() to populate _payload